dbt/adapters/base/relation.py

```diff
@@ -232,17 +233,18 @@ def quoted(self, identifier):
         )
 
     @staticmethod
-    def add_ephemeral_prefix(name: str):
-        return f"__dbt__cte__{name}"
+    def add_ephemeral_prefix(alias: str):
+        return re.sub(r"\W+", "", f"__dbt__cte__{alias}")
```

Comment:
If we do a regex substitution like this, then we could run into issues with ephemeral models like these:
1. `test_#_table`
1. `test_@_table`

If a third model tried to reference both of them, then there would be a name collision since they would both resolve to the same CTE name: `test__table` 💥

So I think we'd be better off **not** altering the value and leaving it as-is.

Reply:
Great point, I removed this logic in e5c1b46807ca939548f2d503a5e6effdcc91ba86.